shopfloor/services/delivery.py

```diff
@@ -90,9 +347,25 @@ def set_qty_done_pack(self, picking_id, package_id):
         Transitions:
         * deliver: always return here with updated data
         """
-        return self._response()
+        picking = self.env["stock.picking"].browse(picking_id).exists()
+        if picking:
+            response = self._check_picking_status(picking)
+            if response:
+                return response
+        else:
+            return self._response_for_deliver(
+                message=self.msg_store.stock_picking_not_found()
+            )
+        package = self.env["stock.quant.package"].browse(package_id).exists()
+        if package:
+            response = self._deliver_package(picking, package)
+            self._action_picking_done(picking)
```

Comment:
`_action_picking_done` to move in `_deliver_package` so it's called too when we scan a package